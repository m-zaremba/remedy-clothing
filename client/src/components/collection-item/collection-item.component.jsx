import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { addItem } from '../../redux/cart/cart.actions';

import {
  CollectionItemContainer,
  BackgroundImage,
  CollectionItemButton,
  CollectionItemFooterContainer,
  FooterItemName,
  FooterItemPrice,
} from './collection-item.styles';

export const CollectionItem = ({ item, addItem }) => {
  const { name, price, imageUrl } = item;
  return (
    <CollectionItemContainer>
      <BackgroundImage imageUrl={imageUrl} />
      <CollectionItemFooterContainer>
        <FooterItemName>{name}</FooterItemName>
        <FooterItemPrice>{price}</FooterItemPrice>
      </CollectionItemFooterContainer>
      <CollectionItemButton inverted onClick={() => addItem(item)}>
        Add to cart
      </CollectionItemButton>
    </CollectionItemContainer>
  );
};

CollectionItem.propTypes = {
  item: PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
    imageUrl: PropTypes.string,
    price: PropTypes.number,
  }).isRequired,
  addItem: PropTypes.func.isRequired,
};

const mapDispatchToProps = (dispatch) => ({
  addItem: (item) => dispatch(addItem(item)),
});

export default connect(null, mapDispatchToProps)(CollectionItem);
