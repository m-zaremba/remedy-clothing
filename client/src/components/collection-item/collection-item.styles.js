import styled from 'styled-components';
import CustomButton from '../custom-button/custom-button.component';

export const CollectionItemContainer = styled.div`
  width: 22vw;
  display: flex;
  flex-direction: column;
  height: 350px;
  align-items: center;
  position: relative;
  margin-bottom: 30px;

  @media screen and (max-width: 800px) {
    width: 80vw;
  }
`;

export const BackgroundImage = styled.div`
  width: 100%;
  height: 95%;
  background-size: cover;
  background-position: center;
  margin-bottom: 5px;
  background-image: ${({ imageUrl }) => `url(${imageUrl})`};

  ${CollectionItemContainer}:hover & {
    opacity: 0.8;
  }
`;

export const CollectionItemButton = styled(CustomButton)`
  width: 80%;
  opacity: 0.7;
  position: absolute;
  top: 255px;
  display: none;

  ${CollectionItemContainer}:hover & {
    opacity: 0.85;
    display: flex;
  }

  @media screen and (max-width: 800px) {
    display: flex;
    min-width: unset;
    padding: 0 10px;
    opacity: 0.9;
  }
`;

export const CollectionItemFooterContainer = styled.div`
  width: 100%;
  height: 5%;
  display: flex;
  justify-content: space-between;
  font-size: 18px;
`;

export const FooterItemName = styled.span`
  width: 90%;
  margin-bottom: 15px;
`;

export const FooterItemPrice = styled.span`
  width: 10%;
`;
