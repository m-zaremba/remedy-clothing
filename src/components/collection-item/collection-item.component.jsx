import React from 'react';
import { connect } from 'react-redux';
import './collection-item.styles.scss';
import PropTypes from 'prop-types';
import CustomButton from '../custom-button/custom-button.component';
import { addItem } from '../../redux/cart/cart.actions';

const CollectionItem = ({ item, addItem }) => {
  const { name, price, imageUrl } = item;
  return (
    <div className="collection-item">
      <div className="image" style={{ backgroundImage: `url(${imageUrl})` }} />
      <div className="collection-footer">
        <span className="name">{name}</span>
        <span className="price">{price}</span>
      </div>
      <CustomButton inverted onClick={() => addItem(item)}>
        Add to cart
      </CustomButton>
    </div>
  );
};

CollectionItem.propTypes = {
  item: PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
    imageUrl: PropTypes.string,
    price: PropTypes.number,
  }).isRequired,
  addItem: PropTypes.func.isRequired,
};

const mapDispatchToProps = (dispatch) => ({
  addItem: (item) => dispatch(addItem(item)),
});

export default connect(null, mapDispatchToProps)(CollectionItem);
