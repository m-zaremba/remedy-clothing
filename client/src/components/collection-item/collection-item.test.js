import React from 'react';
import { shallow } from 'enzyme';

import { CollectionItem } from './collection-item.component';

describe('testing CollectionItem component', () => {
  let wrapper;
  let mockAddItem;
  const mockImageUrl = 'www.testImage.com';
  const mockName = 'black hat';
  const mockPrice = 10;

  beforeEach(() => {
    mockAddItem = jest.fn();

    const mockProps = {
      item: {
        imageUrl: mockImageUrl,
        price: mockPrice,
        name: mockName,
      },
      addItem: mockAddItem,
    };

    wrapper = shallow(<CollectionItem {...mockProps} />);
  });

  it('should render CollectionItem component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call addItem on AddButton click', () => {
    wrapper.find('AddButton').simulate('click');

    expect(mockAddItem).toHaveBeenCalled();
  });

  it('should pass imageUrl as a prop to BackgroundImage component', () => {
    expect(wrapper.find('BackgroundImage').prop('imageUrl')).toBe(mockImageUrl);
  });

  it('should render name value in NameContainer', () => {
    expect(wrapper.find('FooterItemName').text()).toBe(mockName);
  });

  it('should render price value in PriceContainer', () => {
    const price = Number(wrapper.find('FooterItemPrice').text());
    expect(price).toBe(mockPrice);
  });
});
