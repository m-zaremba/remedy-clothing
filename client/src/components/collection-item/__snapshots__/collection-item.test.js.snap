// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing CollectionItem component should render CollectionItem component 1`] = `ShallowWrapper {}`;
